src/Kiota.Builder/Refiners/CodeUsingDeclarationNameComparer.cs

```diff
@@ -8,12 +9,19 @@
 namespace Kiota.Builder.Refiners;
 public class CodeUsingDeclarationNameComparer : IEqualityComparer<CodeUsing>
 {
-    public bool Equals(CodeUsing? x, CodeUsing? y) =>
-        x == null && y == null ||
-        x != null &&
-        y != null &&
-        GetHashCode(x) == GetHashCode(y);
-    public int GetHashCode([DisallowNull] CodeUsing obj) =>
-        (string.IsNullOrEmpty(obj?.Name) ? 0 : obj.Name.GetHashCode(OrdinalIgnoreCase)) * 7 +
-        (string.IsNullOrEmpty(obj?.Declaration?.Name) ? 0 : obj.Declaration.Name.GetHashCode(OrdinalIgnoreCase));
+    public bool Equals(CodeUsing? x, CodeUsing? y)
+    {
+        if (x is null || y is null) return object.Equals(x, y);
+        return string.Equals(x.Name, y.Name, OrdinalIgnoreCase)
```

Comment:
same comment about using the same comparer instance for both the equals and GetHashCode implementations.
Not saying we need to make it customizable via an additional constructor.